device_calendar/lib/src/models/result.dart

```diff
@@ -5,5 +5,11 @@ class Result<T> {
   T data;
   List<String> errorMessages = new List<String>();
 
+    // TODO: Change to allow constructor parameter
   Result([this.data]);
+
```

Comment:
1. Make IsSuccess flag dependant on Data not being null and error messages being empty
2. Change constructor to be parameter free